### PR TITLE
Adding Ballot.sol, testVoting.js

### PR DESCRIPTION
**Pull request by meghana87**
Please review my work.

### CODE REVIEW

**Comment by nrchandan** [comment]
Unrelated to the lms app.

File: contracts/Ballot.sol [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	struct Candidate {
+		address candidate_account;
+		uint votecount;
+	}
+	uint public numCandidates;
+
+
+	function Ballot(address[] addresses_) {
+		for (uint i=0; i < addresses_.length; i++) {
+			Ballot[++numCandidates] = Candidate(addresses_[i],0);
```
Comment (nrchandan) [inline contracts/Ballot.sol]: Why ++numCandidates and not numCandidates++

Comment (nrchandan) [inline contracts/Ballot.sol]: Why a trailing underscore in addresses_? Is "addresses" a keyword?

File: contracts/Ballot.sol [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		}
+	}
+
+	function TotalVotesReceived(address account) returns (uint) {
```
Comment (nrchandan) [inline contracts/Ballot.sol]: Should this function be declared constant?

File: contracts/Ballot.sol [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				Ballot[i].votecount++;
+			}
+		}
+	}
+
+	function TotalCandidates() constant returns (uint) {
+		return numCandidates;
+	}
+
+	function validCandidate(address account) returns (bool) {
```
Comment (nrchandan) [inline contracts/Ballot.sol]: This function's name is camelCase while the others are TitleCase. Stick to one. What does the style guide for Solidity have to say about the naming convention?

Comment (nrchandan) [inline contracts/Ballot.sol]: Should this be a constant function?

File: contracts/Ballot.sol [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		address winner;
+		for (uint i=1; i <= numCandidates; i++) {
+			if (Ballot[i].votecount > winning_count) {
+				winning_count = Ballot[i].votecount;
+				winner = Ballot[i].candidate_account;
+			}
+		}
+		return winner;
+	}
+}
```
Comment (nrchandan) [inline contracts/Ballot.sol]: Do you see any improvement or shortcoming in this implementation?

File: test/testVoting.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    describe('FindWinner', function() {
+    	it('should return the winning address', async function() {
+    		await ballot.VoteFor(accounts[0]);
+        	await ballot.VoteFor(accounts[0]);
+        	await ballot.VoteFor(0x0);
+        	await ballot.VoteFor(0x0);
+        	await ballot.VoteFor(0x0);
+    		let winner = await ballot.FindWinner.call();
+    		assert.equal(winner.valueOf(),0x0, 'Found a winner.');
```
Comment (nrchandan) [inline test/testVoting.js]: Use coding standards - there should be a single space after each comma. Ideally, your code editor should help you identify and fix such issues. I know PyCharm does that. Not sure about your editor.

File: contracts/Ballot.sol [hunk 6]
```diff
@@ -0,0 +1,61 @@
+pragma solidity ^0.4.0;
+
+
+contract Ballot {
+	mapping (uint => Candidate) public Ballot;
+	struct Candidate {
+		address candidate_account;
+		uint votecount;
```
Comment (neeldhwaj) [inline contracts/Ballot.sol]: Use consistent naming convention.

File: contracts/Ballot.sol (hunk 1 above)
Comment (meghana87) [inline contracts/Ballot.sol]: 1)++numCandidates gives the expression value after change while numCandidates++ gives the expression value before change. So, now updated the code as per the logic.
2)Yes..trailing underscore is not needed here as "addresses" is not a built-in keyword.

File: contracts/Ballot.sol (hunk 2 above)
Comment (meghana87) [inline contracts/Ballot.sol]: Yes..this function can be declared "constant" as we are not changing the state of "votecount" variable.

File: contracts/Ballot.sol (hunk 3 above)
Comment (meghana87) [inline contracts/Ballot.sol]: 1) Updated the code by following one of the naming convention where we use one uppercase letter while naming functions.
2) Yes..the above function can also be declared "constant" as we are not changing the state of return value(bool).

File: contracts/Ballot.sol (hunk 4 above)
Comment (meghana87) [inline contracts/Ballot.sol]: Yes. I implemented the above functionality by changing the logic slightly. Please review.

File: test/testVoting.js (hunk 5 above)
Comment (meghana87) [inline test/testVoting.js]: I was using "sublime" so far which did not point out such issues in my code. However, now updated the code as per the coding standards.

**Pull request closed without merging**